src/output.c

```diff
@@ -141,6 +141,10 @@ output_destroy_notify(struct wl_listener *listener, void *data)
 	 * destroyed automatically when the wlr_output is destroyed
 	 */
 	free(output);
+
+	if (server->fallback_output && wl_list_length(&server->outputs) == 0) {
+		output_add_virtual(server, "HEADLESS");
+	}
```

Comment:
`s/wl_list_length/wl_list_empty/`
However, this might actually fail because there may be outputs that are disabled or used only for DRM leasing.

We likely need something like this instead:
```c
bool output_available(struct server *server)
{
	wl_list_for_each(output, &server->outputs, link) {
		if (output_is_usable(output)) {
			return true;
		}
	}
	return false;
}
```